.pipelines/ci.yml

```diff
@@ -20,78 +20,130 @@ pr:
+        editedFiles=($editedFiles)
+
+        for (( i=0; i<${#editedFiles[@]}; i++ ))
+        do
+            case ${editedFiles[$i]} in
+            *.go)
+                echo "##vso[task.setvariable variable=IsGolangChanges]True"
+                ;;
+
+            *.py)
+                echo "##vso[task.setvariable variable=IsPythonChanges]True"
+                ;;
+
+            *.tsx)
+                echo "##vso[task.setvariable variable=IsTypescriptChanges]True" 
+                ;;
+            esac
+        done
+
+        IFS=$SAVEIFS   # Restore original IFS
+
+- stage: Unit_Tests_And_Linting
+  jobs:
+  - job: Python_Unit_Tests
+    pool:
+      name: ARO-CI
+    steps:
+    - template: ./templates/template-checkout.yml
+    - script: |
+        set -xe
+        make test-python
+        [[ -z "$(git status -s)" ]]
+      displayName: 🧪Run Python Unit Tests
+    condition: and(succeeded(), eq(variables['IsPythonChanges'], True))
+
+  - job: Golang_Unit_Tests
+    pool:
+      name: ARO-CI
+      demands: go-1.16
+    steps:
+    - template: ./templates/template-checkout.yml
+
+    - script: |
+        set -xe
+        make generate
+        [[ -z "$(git status -s)" ]]
+      displayName: ⚙️ Run Golang code generate
+
+    - script: |
+        set -xe
+        make validate-go
+        [[ -z "$(git status -s)" ]]
+      displayName: 🕵️ Validate Golang code
+
+    - script: |
+        set -xe
+        make lint-go
+        [[ -z "$(git status -s)" ]]
+      displayName: 🕵️ Lint Golang code
+
+    - script: |
+        set -xe
+        make build-all
+        [[ -z "$(git status -s)" ]]
+      displayName: 🕵️ Build Golang code
+
+    - script: |
+        set -xe
+        make unit-test-go
+      displayName: 🧪 Run Golang unit tests
+
+    - script: |
+        set -xe
+        make validate-fips
+      displayName: 🕵️ Validate FIPS
+
+    - task: PublishTestResults@2
+      displayName: 📊 Publish tests results
+      inputs:
+        testResultsFiles: $(System.DefaultWorkingDirectory)/**/report.xml
+      condition: succeededOrFailed()
+
+    - script: |
+        go run ./vendor/github.com/axw/gocov/gocov/*.go convert cover.out > coverage.json
+        go run ./vendor/github.com/AlekSi/gocov-xml/gocov-xml.go < coverage.json > coverage.xml
+      displayName: ⚙️ Process Reports
+      condition: succeededOrFailed()
+
+    - task: PublishCodeCoverageResults@1
+      displayName: 📈 Publish code coverage
+      inputs:
+        codeCoverageTool: Cobertura
+        summaryFileLocation: $(System.DefaultWorkingDirectory)/**/coverage.xml
+        failIfCoverageEmpty: false
+      condition: succeededOrFailed()
+    condition: and(succeeded(), eq(variables['IsGolangChanges'], True))
+
+  - job: Lint_Admin_Portal
+    pool:
+      name: ARO-CI
+    steps:
+    - script: |
+        set -xe
+        make lint-admin-portal
+      displayName: 🧹 Lint Admin Portal
+    condition: and(succeeded(), eq(variables['IsTypescriptChanges'], True))
```

Comment:
Should we be running a `make build-portal && make generate` here as well if there are typescript changes?

Reply:
We would also have to commit the bindata generated from those changes. Which is something we can potentially do in the future. The lint itself only checks the commited code in the portal directory